metricsbp/statsd.go

```diff
@@ -159,34 +159,46 @@ func NewStatsd(ctx context.Context, cfg StatsdConfig) *Statsd {
 	return st
 }
 
-// Counter returns a counter metrics to the name.
-//
-// It uses the DefaultCounterSampleRate used to create Statsd object.
-// If you need a different sample rate,
-// you could use st.Statsd.NewCounter instead.
+// Counter returns a counter metrics to the name,
+// with sample rate inherited from StatsdConfig.
 func (st *Statsd) Counter(name string) metrics.Counter {
 	st = st.fallback()
-	return st.Statsd.NewCounter(name, st.counterSampleRate)
+	counter := st.Statsd.NewCounter(name, st.counterSampleRate)
+	if st.counterSampleRate >= 1 {
```

Comment:
Should we also do this if `st.counterSampleRate == 0`?

Reply:
That will make things slightly faster, but then again I don't think we actually use zero sample rate in real world so this sounds like a premature optimization. The current code still works when sample rate is 0, it's just slower.

Reply:
I would only argue that `0` is the default value, so it might be worth adding the optimization just because of that?  That said:

1. I'm not sure how often that'll come up
2. I'm not sure how much of a gain it actually is

Reply:
1 is the actual default value now (since last change)

Reply:
Welp, I think it’s more than safe to ignore then 😅